packages/flutter_tools/test/general.shard/plugin_parsing_test.dart

```diff
@@ -290,6 +290,29 @@ void main() {
     ]);
   });
 
+  testWithoutContext('Web plugin tool exits if fileName field missing', () {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    const String pluginYamlRaw =
+      'platforms:\n'
+      ' web:\n'
+      '  pluginClass: WebSamplePlugin\n';
+
+    final YamlMap pluginYaml = loadYaml(pluginYamlRaw) as YamlMap;
+    expect(
+      () => Plugin.fromYaml(
+        _kTestPluginName,
+        _kTestPluginPath,
+        pluginYaml,
+        null,
+        const <String>[],
+        fileSystem: fileSystem,
+      ),
+      throwsToolExit(
```

Comment:
Should there be an additional test for this case? `throwToolExit('The plugin `$name` is missing the required field `pluginClass` in pubspec.yaml');`

Reply:
If I simply exlude that field, we actually don't even reach this method, because of previous parsing. I left this toolexit though in case there's some other weird case in which we CAN reach this method (I got confused tracing the previous logic)

Reply:
Fair enough